src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -76,6 +86,12 @@ public void Apply(BsonClassMap classMap)
                 // if any constructors were mapped by this convention then map all the properties also
                 foreach (var property in properties)
                 {
+                    if (property.DeclaringType != classMap.ClassType)
```

Comment:
In line 48 we're getting all properties in the class (including properties from the base classes), it's required because constructor can pass parameters directly to the base class via `: base(...)`, but here we won't save properties which don't belong to the current level. The skipped properies will be processed when we will process the base type further

Reply:
Ack

Reply:
> when we will process the base type further

I think the base type has **already** been mapped at this point.

Reply:
Alternative refactoring to consider:

    foreach (var property in properties.Where(p => p.DeclaringType == classMap.ClassType))

but either way is fine.

Reply:
>I think the base type has already been mapped at this point.

nope, we go down the hierarchy from high-level class in the inheritance hierarchy to base classes

Reply:
>Alternative refactoring to consider:

I considered this refactoring but decided that the current way is more flexible, at least it's possible to set breakpoints between "continue" brackets. 
I can apply this suggestion, but I think the current way is also good looking, so let me know if you feel that your suggestion is better than the currently added